src/jackdaw/client.clj

```diff
@@ -57,13 +57,13 @@
   either [RecordMetdata, nil] or [nil, Exception] respectively if the
   record was sent or if an exception was encountered."
   ([producer record]
-   (-> (.send ^Producer producer ^ProducerRecord record)
-       deref jd/datafy delay))
+   (let [send-future (.send ^Producer producer ^ProducerRecord record)]
+     (delay (jd/datafy @send-future))))
```

Comment:
Move the `send!` call outside of the `delay` macro, so that the send call is actually made here and not delayed until the return value of this function is deref'd